### PR TITLE
Improve install pci, search image in snapshot + check if instance is …

### PR DESCRIPTION
**Pull request by guillaume-gambs**
Hi again ;)

I add some improvement into public cloud installation.
Now we can install an instance based on snapshot ( for example if you import your qcow2 image in ovh openstack )
Check before create instance if an instance with the same name and in same datacenter, is already created, and if this instance exist, skip the command of install, except if we want force a new installation.

### CODE REVIEW

**Comment by atrawog** [comment]
I just gave this PR a quick try and it fixed my issue that duplicated cloud instance get created with the same name every time I re-run the playbook.

File: plugins/modules/public_cloud_instance.py [hunk 1]
```diff
@@ -94,11 +109,38 @@ def run_module():
...
+                    reinstall_result = client.post(
+                                        '/cloud/project/%s/instance/%s/reinstall' % (service_name,instance_id),
+                                         imageId=image_id
+                                         )
+                    module.exit_json(changed=True, **reinstall_result)
+
+                except APIError as api_error:
+                    module.fail_json(msg="Failed to call OVH API: {0}".format(api_error))
+
+            module.exit_json(changed=False,**instance_details)
```
Comment (Skunnyk) [inline plugins/modules/public_cloud_instance.py:143]: Maybe we can exit with a message ? Like `module.exit_json(changed=False,msg="Instance {} is already installed".format(name)` ?

**Comment by Skunnyk** [comment]
Thank you ! 
Can you please run  `flake8` for formatting stuff ? (I need to add some github action to do that magically)

File: plugins/modules/installation_template.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def run_module():
             try:
                 client.delete('/me/installationTemplate/%s' % src_template)
 
-                module.exit_json(msg="Template {} succesfully deleted".format(src_template, changed=True))
+                module.exit_json(msg="Template {} succesfully deleted".format(src_template), changed=True)
```
Comment (Skunnyk) [inline plugins/modules/installation_template.py:87]: Wow, good catch ^^"

**Comment by Skunnyk** [comment]
Yai, thank you Guillaume !

**Pull request merged**